pkg/webhooks/workload_webhook.go

```diff
@@ -182,6 +184,87 @@ func validateContainer(c *corev1.Container, path *field.Path) field.ErrorList {
 	return allErrs
 }
 
+func validateAdmissionChecks(obj *kueue.Workload, basePath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	for i := range obj.Status.AdmissionChecks {
+		admissionChecksPath := basePath.Index(i)
+		ac := &obj.Status.AdmissionChecks[i]
+		allErrs = append(allErrs, validatePodSetUpdates(ac, obj, admissionChecksPath.Child("podSetUpdates"))...)
+	}
+	return allErrs
+}
+
+func validatePodSetUpdates(acs *kueue.AdmissionCheckState, obj *kueue.Workload, basePath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	knowPodSets := make(map[string]struct{}, len(obj.Spec.PodSets))
+	knowPodSetNames := make([]string, len(obj.Spec.PodSets))
+	for i := range obj.Spec.PodSets {
+		name := obj.Spec.PodSets[i].Name
+		knowPodSets[name] = struct{}{}
+		knowPodSetNames = append(knowPodSetNames, name)
+	}
+	for i := range acs.PodSetUpdates {
+		psu := &acs.PodSetUpdates[i]
+		psuPath := basePath.Index(i)
+		for _, msg := range validation.IsDNS1123Label(psu.Name) {
+			allErrs = append(allErrs, field.Invalid(psuPath.Child("name"), psu.Name, msg))
+		}
+		if _, found := knowPodSets[psu.Name]; !found {
+			allErrs = append(allErrs, field.NotSupported(psuPath.Child("name"), psu.Name, knowPodSetNames))
+		}
+		allErrs = append(allErrs, validateTolerations(psu.Tolerations, psuPath.Child("tolerations"))...)
+		allErrs = append(allErrs, apivalidation.ValidateAnnotations(psu.Annotations, psuPath.Child("annotations"))...)
+		allErrs = append(allErrs, metav1validation.ValidateLabels(psu.Labels, psuPath.Child("labels"))...)
+	}
+	return allErrs
+}
+
+// validateTolerations is extracted from git.k8s.io/kubernetes/pkg/apis/core/validation/validation.go
+func validateTolerations(tolerations []corev1.Toleration, fldPath *field.Path) field.ErrorList {
```

Comment:
Oh, I don't really like this but found we have also other copy-paste places like this. https://github.com/kubernetes-sigs/kueue/blob/525098b838eb28ee241cf1aaee5e53f161d30a0c/pkg/webhooks/resourceflavor_webhook.go#L134. Is there no way to depend on the `git.k8s.io/kubernetes/pkg/apis/core/validation/` package?

Reply:
As I know, libraries are published as standalone modules and can be referenced directly. I've never meet validation library as standalone module. We can consider depending on k8s.io/kubernetes directly but it's not recommended: https://github.com/kubernetes/kubernetes/issues/79384#issuecomment-505627280

Reply:
Ok, I would suggest adding the link to the comment so that we have a pointer.

Reply:
added comment https://github.com/kubernetes-sigs/kueue/pull/1220/commits/dd3dde19094c7988fcd3440d59aaf40d1dafee6b